core/db_classes/EE_Transaction.class.php

```diff
@@ -1691,4 +1691,25 @@ public function verify_abandoned_transaction_status()
             }
         }
     }
+
+
+    /**
+     * @since $VID:$
+     * @throws EE_Error
+     * @throws InvalidArgumentException
+     * @throws InvalidDataTypeException
+     * @throws InvalidInterfaceException
+     * @throws ReflectionException
+     * @throws RuntimeException
+     */
+    public function recalculateLineItems()
+    {
+        $total_line_item = $this->total_line_item(false);
+        $success = false;
+        if ($total_line_item instanceof EE_Line_Item) {
+            EEH_Line_Item::resetIsTaxableForTickets($total_line_item);
+            $success = EEH_Line_Item::apply_taxes($total_line_item, true);
+        }
+        return $success;
```

Comment:
slightly leaner version just for demonstration purposes but what you have is perfectly acceptable and even preferential to some

```suggestion
        $total_line_item = $this->total_line_item(false);
        if ($total_line_item instanceof EE_Line_Item) {
            EEH_Line_Item::resetIsTaxableForTickets($total_line_item);
            return EEH_Line_Item::apply_taxes($total_line_item, true);
        }
        return false;
```

Reply:
Lol I wasn't going to say anything :)